nautobot/wireless/models.py

```diff
@@ -0,0 +1,253 @@
+    authentication = models.CharField(
+        max_length=CHARFIELD_MAX_LENGTH,
+        choices=WirelessAuthTypeChoices,
+    )
+    secrets_group = models.ForeignKey(
+        to="extras.SecretsGroup",
+        on_delete=models.SET_NULL,
+        blank=True,
+        null=True,
+    )
+    hidden = models.BooleanField(default=False)
+
+    class Meta:
+        ordering = ["name"]
+
+    def __str__(self):
+        return self.name
+
+
+@extras_features(
+    "custom_links",
+    "custom_validators",
+    "export_templates",
+    "graphql",
+)
+class AccessPointGroupWirelessNetworkAssignment(BaseModel):
+    """
+    An AccessPointGroupWirelessNetworkAssignment represents the assignment of a WirelessNetwork to an AccessPointGroup.
+    """
+
+    access_point_group = models.ForeignKey(
+        to="wireless.AccessPointGroup",
+        on_delete=models.CASCADE,
+        related_name="wireless_network_assignments",
+    )
+    wireless_network = models.ForeignKey(
+        to="wireless.WirelessNetwork",
+        on_delete=models.CASCADE,
+        related_name="access_point_group_assignments",
+    )
+    vlan = models.ForeignKey(
+        to="ipam.VLAN",
+        on_delete=models.PROTECT,
+        related_name="access_point_group_wireless_network_assignments",
+        blank=True,
+        null=True,
+    )
+    is_metadata_associable_model = False
+
+    class Meta:
+        unique_together = ["access_point_group", "wireless_network"]
+        ordering = ["access_point_group", "wireless_network"]
+
+    def __str__(self):
+        return f"{self.access_point_group}: {self.wireless_network}"
+
+
+@extras_features(
+    "custom_links",
+    "custom_validators",
+    "export_templates",
+    "graphql",
+)
+class AccessPointGroupRadioProfile(BaseModel):
+    """
+    An AccessPointGroupRadioProfile represents the assignment of a RadioProfile to an AccessPointGroup.
+    """
+
+    access_point_group = models.ForeignKey(
+        to="wireless.AccessPointGroup",
+        on_delete=models.CASCADE,
+        related_name="radio_profile_assignments",
+    )
+    radio_profile = models.ForeignKey(
+        to="wireless.RadioProfile",
+        on_delete=models.CASCADE,
+        related_name="access_point_group_assignments",
+    )
+    is_metadata_associable_model = False
+
+    class Meta:
+        unique_together = ["access_point_group", "radio_profile"]
+        ordering = ["access_point_group", "radio_profile"]
+
+    def __str__(self):
+        return f"{self.access_point_group}: {self.radio_profile}"
+
+
+@extras_features(
+    "custom_links",
+    "custom_validators",
+    "export_templates",
+    "graphql",
+)
+class AccessPointGroupDevicesAssignment(BaseModel):
+    """
+    An AccessPointGroupDevicesAssignment represents the assignment of a Device to an AccessPointGroup.
+    """
+
+    access_point_group = models.ForeignKey(
+        to="wireless.AccessPointGroup",
+        on_delete=models.CASCADE,
+        related_name="access_points",
```

Comment:
Should this be related_name="devices" or "related_name="device_assignments""? Sorry if Im wrong im not sure.

Reply:
Doesn't this mean that `AccessPointGroup.access_points.all()` would return an `AccessPointGroupDevicesAssignment` manager? Shouldn't `AccessPointGroup.access_points.all()` return a `Device` manager?

Actually, aren't we missing the M2M for this through table?

Reply:
I am not sure, I will have to check the ORM.

But, yes, at some point I deleted the M2M.

Reply:
Fixed it, I was missing the M2M call for all three through tables. `AccessPointGroup.access_points.all()` now returns the proper manager.